src/lithium/tests/test_lithium.py

```diff
@@ -18,7 +18,7 @@
 import tempfile
 import unittest
 
-import lithium  # pylint: disable=relative-import
+from .. import reducer
```

Comment:
This should be an absolute import (`import lithium` or `from lithium import reducer`, with the `__init__.py` change I suggested above you could do either).

When I went looking for a reference I came across this: https://docs.pytest.org/en/latest/goodpractices.html#choosing-a-test-layout-import-rules

This advocates for placing `tests` at the top level, not under `src/lithium`, and I think their reasons make sense. From the [blog post](https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure) (linked from pytest docs above):
> You'll notice that I don't include the tests in the installed packages. Because:
> * Module discovery tools will trip over your test modules. Strange things usually happen in test module. The help builtin does module discovery. E.g.:
>   ```
>   >>> help('modules')
>   Please wait a moment while I gather a list of all available modules...
>   
>    __future__          antigravity         html                select
>   ...
>   ```
>
> * Tests usually require additional dependencies to run, so they aren't useful by their own - you can't run them directly.
> * Tests are concerned with development, not usage.
> * It's extremely unlikely that the user of the library will run the tests instead of the library's developer. E.g.: you don't run the tests for Django while testing your apps - Django is already tested.

Note that `__init__.py` is not needed unless we have duplicate test names.